tests/perf-system/analyzer/analyzer.py

```diff
@@ -1,348 +1,490 @@
+    ):
+        """
+        Starting from minimum latency with ms_separator
+        step split the ms latency list in buckets
+        and plots the highest_vals top buckets
+        """
+        max_latency = max(self.ms_latency_list)
+        min_latency = min(self.ms_latency_list)
+
+        if max_latency < ms_separator:
+            LOG.remove()
+            LOG.add(
+                sys.stdout,
+                format="<red>[ERROR]:</red> {message}",
+            )
+
+            LOG.error(
+                f"Latency values are less than {ms_separator}, cannot produce latency distribution graph"
+            )
+            return
+
+        bins_number = (
+            int(
+                (max_latency - min_latency) // ms_separator
+                + bool((max_latency - min_latency) % ms_separator)
+            )
+            + 1
+        )
+
+        counts = [0] * bins_number
+        bins = [min_latency]
+        bin_val = min_latency
+        for _ in range(bins_number):
+            bin_val += ms_separator
+            bins.append(bin_val)
+
+        for lat in self.ms_latency_list:
+            counts[
+                int(
+                    (lat - min_latency) // ms_separator
+                    + bool((lat - min_latency) % ms_separator)
+                )
+            ] += 1
+
+        top_bins = []
+        top_counts = []
+        min_count = sorted(counts)[-highest_vals]
+        for ind in range(len(counts)):
+            if counts[ind] >= min_count:
+                top_bins.append(round(bins[ind], 3))
+                top_counts.append(counts[ind])
+
+        x_axis = range(len(top_bins))
+        plt.figure()
+        fig, ax = plt.subplots()
+        ax.bar(x_axis, top_counts, 0.9, align="center")
+        ax.set_xticks(x_axis)
+        ax.set_xticklabels(top_bins, rotation=25)
+        plt.ylabel("Count")
+        plt.xlabel("Latency (ms)")
+        fig.subplots_adjust(bottom=0.2)
+        plt.savefig(results_directory / "latency_distribution.pdf")
+        plt.close()
+
+
+def get_df_from_parquet_file(input_file: Path):
+    return pd.read_parquet(input_file)
+
+
+def prettify_graphs():
+    fontsize = 9
+    params = {
+        "axes.labelsize": fontsize,
+        "font.size": fontsize,
+        "legend.fontsize": fontsize,
+        "xtick.labelsize": fontsize,
+        "ytick.labelsize": fontsize,
+        "figure.figsize": DEFAULT_FIGURE_SIZE,
+        "xtick.direction": "in",
+        "ytick.direction": "in",
+        "xtick.top": True,
+        "ytick.right": True,
+        "xtick.minor.visible": True,
+        "ytick.minor.visible": True,
+        "lines.linewidth": 2,
+        "legend.frameon": False,
+        "axes.grid": False,
+        "savefig.bbox": "tight",
+    }
+    plt.rcParams.update(params)
+
+
+def default_analysis(
+    input_file: Path,
+    send_file: Path,
+    response_file: Path,
+    results_directory=Path("."),
```

Comment:
Previously plots were always saved to the local directory, this is now customizable